executor/src/main/java/com/redhat/service/bridge/executor/actions/kafka/KafkaTopicActionInvokerFactory.java

```diff
@@ -0,0 +1,67 @@
+package com.redhat.service.bridge.executor.actions.kafka;
+
+import java.util.Set;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+
+import javax.enterprise.context.ApplicationScoped;
+import javax.inject.Inject;
+
+import org.apache.kafka.clients.admin.AdminClient;
+import org.eclipse.microprofile.reactive.messaging.Channel;
+import org.eclipse.microprofile.reactive.messaging.Emitter;
+
+import com.redhat.service.bridge.executor.actions.ActionInvoker;
+import com.redhat.service.bridge.executor.actions.ActionInvokerException;
+import com.redhat.service.bridge.executor.actions.ActionInvokerFactory;
+import com.redhat.service.bridge.infra.models.actions.BaseAction;
+import com.redhat.service.bridge.infra.models.actions.KafkaTopicAction;
+import com.redhat.service.bridge.infra.models.dto.ProcessorDTO;
+
+import io.cloudevents.CloudEvent;
+
+@ApplicationScoped
+public class KafkaTopicActionInvokerFactory implements ActionInvokerFactory {
+
+    static final long DEFAULT_LIST_TOPICS_TIMEOUT = 10L;
+
+    static final TimeUnit DEFAULT_LIST_TOPICS_TIMEUNIT = TimeUnit.SECONDS;
+
+    @Channel("actions-out")
+    Emitter<CloudEvent> emitter;
+
+    @Inject
+    AdminClient adminClient;
+
+    @Override
+    public boolean accepts(BaseAction baseAction) {
+        if (baseAction != null) {
+            return KafkaTopicAction.KAFKA_ACTION_TYPE.equals(baseAction.getType());
+        }
+        return false;
+    }
+
+    @Override
+    public ActionInvoker build(ProcessorDTO processor, BaseAction baseAction) {
+
+        String requiredTopic = baseAction.getParameters().get(KafkaTopicAction.KAFKA_ACTION_TOPIC_PARAM);
+        if (requiredTopic == null) {
+            throw new ActionInvokerException(
+                    String.format("There is no topic specified in the parameters for Action on Processor '%s' on Bridge '%s'", processor.getId(), processor.getBridge().getId()));
+        }
+
+        try {
+            Set<String> strings = adminClient.listTopics().names().get(DEFAULT_LIST_TOPICS_TIMEOUT, DEFAULT_LIST_TOPICS_TIMEUNIT);
```

Comment:
I'm not sure about this check: adminApi can be disabled in some Kafka installation and moreover this doesn't prevent user to removed a topic later 🤔

Reply:
I think for this iteration it will help us get things up and running and capture obvious errors early. We also know that we can pretty much guarantee admin API is enabled for MK at this point, so I'd say +1 to keep it for now

Longer-term, I'd say we move it to a `param` on the Action e.g `checkTopicExists=true` which can then make the assumption that we have admin API capabilities from the credentials that the user has provided.

If that works for you, I'll add an issue to the backlog to make the topic check on Action provisioning a configurable thing?